### DISCUSSION TITLE
Is it possible to create default set of attributes for each trace/metrics not from env?

### DISCUSSION THREAD

**Discussion by maxifom** [open-telemetry/opentelemetry-go#2211] [Q&A] [answered]
I can see there is a way to specify service.name and any other attributes using OTEL_SERVICE_NAME and OTEL_RESOURCE_ATTRIBUTES env variables. As I can see, it's global defaults for service_name and other attributes.
Is there a way to set default for attribute, not using env? For e.g I want during the startup of my program to set service.instance.id to uuid4, which I guess is a good case. How do I do that?

**Comment by maxifom** [comment]
I found the solution, it was to create Resource and pass it to exporter/tracer (prometheus/jaeger) for WithResource function.

**Comment by Aneurysm9** [comment, marked as answer]
These can be done by constructing a [`*resource.Resource`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v1.0.0-RC2/resource#Resource) with the desired attributes and using it to construct a `TracerProvider` using the [`WithResource` option](https://github.com/open-telemetry/opentelemetry-go/blob/sdk/v1.0.0-RC2/sdk/trace/provider.go#L280).
